.github/workflows/ci.yaml

```diff
@@ -0,0 +1,93 @@
+---
+name: CI
+
+on:
+  push:
+    paths-ignore:
+      - README.md
+      - .gitignore
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository_owner }}/comment
+
+jobs:
+  version:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: '0'
+      - name: Get new version
+        id: version_tag
+        uses: anothrNick/github-tag-action@8c8163ef62cf9c4677c8e800f36270af27930f42 # 1.61.0
```

Comment:
@timberhill I have missed the party, however have a question here - do we want to create a tag each time a change is pushed ? 
I'm a bit used to having this the way around, tag can trigger the pipeline, not vice versa.
This might be a good thing to have on `main` or `release/*` branch, not so sure whether this should apply to every branch though

Reply:
Fair enough, yes. This bit, however, only tags the repo, _not_ any images. So what it does is just generate a new version tag for the repo. If the workflow runs _not_ on the `main` branch, the version is generated like `0.4.0-beta0` for instance.

The build stage depends on the version, so if the version is not generated, the image is not built.

Hope that makes sense. This is not necessarily the best way to do this and we can chat if you'd like :)

Reply:
** My bad - I have missed a big highlighted area saying that container registry is free for now
![image](https://user-images.githubusercontent.com/800419/226214313-47b8fefc-c515-4ee7-8c17-685d9cf66811.png)
Please disregard the billing part, still a lot of images too keep :)

Reply:
I'll play around with the options in the workflow for `submission` service & we can maybe continue the discussion as you will be reviewing the PR.

Reply:
Good point about storage though, I have not considered it. Good thing it's free :D

The image tags currently match the git tags. By sha, do you mean the commit sha?
Building on push I thought is quite nice because it allows to pull images from branches for testing if needed.

Reply:
Yes, I was referring to commit SHA, as an example though. I think anything branch related would do. 
It does make sense convenience wise + works well for smaller changes. I am just a bit unused to the setup and prefer to publish new images by an explicit action.

Reply:
We _could_ only build on the main branch though, it's not a problem. Then we would just push an image by merging a PR.

The docker-compose in dev-stack repo builds its own images locally, so that would be okay. With helm charts however, we might need the beta builds if we want to test code off the main branch.

Reply:
But we could also have a workflow that runs on a cronjob and deletes the older images, for instance, any beta build older than 7 days.

Reply:
I didn't mean to suggest building _only_ from `main` + in some cases you _want_ the image before merging the MR. 
My only concern is building on _every_ push - seems excessive. 

E.g. one thing I'm used to - the _publish whatever it is we're building_ to happen when the workflow is triggered by tag being pushed, e.g. 
```
on:
  push:
    tags:
      - '*'
```

Reply:
Right, but currently the tags are pushed on every push  (not commit). Would need to change that (which is fine, just pointing it out)

Reply:
Agreed - managing image retention is another way to do it. 
Might be a good idea regardless of the image publish condition :)